### DISCUSSION TITLE
What method can I use to intercept "textDocument/definition" because I want to customize the peek definition or go to definition

### DISCUSSION THREAD

**Discussion by NightMingGit** [TypeFox/monaco-languageclient#483] [Q&A]
![image](https://user-images.githubusercontent.com/92633021/231622974-0b88b632-aef5-4e19-89fe-5835924a6873.png)

**Comment by NightMingGit** [comment]
What I really want to do is to go to definition across files and I don't know how to do that, right

**Comment by kaisalmen** [comment]
Hi @NightMingGit you have to do some work on the client side to make such things work. For inspiration take a look at the languageclient code used in:
https://github.com/redhat-developer/vscode-java
Especially here:
https://github.com/redhat-developer/vscode-java/blob/master/src/apiManager.ts

**Reply by CGNonofr** [reply]
Note that you can register middlewares on language client and it seems there is one for definitions

but I'm not sure that the proper way to do what they intend to do
